### ISSUE TITLE
Price inclusive of tax having additional tax amount

### ISSUE THREAD

**Issue by jazgraphy** [woocommerce/woocommerce#15439]
### EXPLANATION OF THE ISSUE

I have issue with WooCommerce Tax settings.
I've set my Tax Options for "Prices entered with tax" to "Yes, I will enter prices inclusive of tax".
But the system still add an extra tax amount on top of the price, even for new products. 
For example, product price is $100 inclusive 6% tax, but it shows $106.

### STEPS TO REPRODUCE THE ISSUE

My Tax Options settings as below:
Tax options

Prices entered with tax: Yes, I will enter prices inclusive of tax
Calculate tax based on: Customer shipping address
Shipping tax class: Zero Rate
Rounding: Round tax at subtotal level, instead of rounding per line
Additional tax classes: Zero Rate
Display prices in the shop: Including tax
Display prices during cart and checkout	: Including tax
Price display suffix: inc. GST
Display tax totals: Itemized

### SYSTEM STATUS

<details>
```
Grab the system status report from WooCommerce > System Status and paste it here between the `details` tags.
```
</details>
`
### WordPress Environment ###

Home URL: 
Site URL: 
WC Version: 3.0.7
Log Directory Writable: ✔
WP Version: 4.7.5
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 5.6.30
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.53.1
OpenSSL/1.0.1e

SUHOSIN Installed: –
MySQL Version: ❌ 5.5.54 - We recommend a minimum MySQL version of 5.6. See: WordPress requirements
Max Upload Size: 8 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.0.7
WC Database Prefix: hd_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (25) ###

AfterShip - WooCommerce Tracking: by AfterShip – 1.6.8
Google Analytics Counter Tracker: by WPAdm – 3.5.6
Contact Form 7: by Takayuki Miyoshi – 4.7
Easy Theme and Plugin Upgrades: by Chris Jean – 2.0.0
Flamingo: by Takayuki Miyoshi – 1.5
Import users from CSV with meta: by codection – 1.10.6.4
Nextend Facebook Connect: by Nextendweb – 2.0.2
PW WooCommerce Exclude Free Shipping: by Pimwick
LLC – 1.4

Regenerate Thumbnails: by Alex Mills (Viper007Bond) – 2.2.6
Ultimate WooCommerce Brands: by MagniumThemes – 1.1.3
WooCommerce Tree Table Rate Shipping: by tablerateshipping.com – 1.15.3
WooCommerce Weight Based Shipping +: by dangoodman – 5.0.5
WooCommerce - Store Exporter: by Visser Labs – 1.9
WooCommerce Seamless MOLPay: by MOLPay Tech Team – 2.3.0
WooCommerce PDF Invoices: by Bas Elbers – 2.9.1
WooCommerce Sequential Order Numbers Pro: by WooThemes / SkyVerge – 1.10.0 – 1.11.2 is available
WP All Import - WooCommerce Add-On: by Soflyy – 1.3.3
WooCommerce: by Automattic – 3.0.7
WooSidebars: by WooThemes – 1.4.3
WP All Import Pro: by Soflyy – 4.3.1
WP Sync DB: by Sean Lang – 1.5
WP All Import - Link Cloaking Add-on: by Soflyy – 1.1.0
WP All Import - User Import Add-On: by Soflyy – 1.0.9
YITH WooCommerce Points and Rewards: by YTHEMES – 1.1.0
YITH WooCommerce Wishlist: by YITHEMES – 2.1.2

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: MYR (RM)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Shop base: #63 - /shop/
Cart: #62 - /cart/
Checkout: #61 - /checkout/
My account: #60 - /my-account/

### Theme ###

Name: Flatsome
Version: 3.3.5
Author URL: http://www.uxthemes.com/
Child Theme: ❌ – If you're modifying WooCommerce on a parent theme you didn't build personally
then we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-product_cat.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/orderby.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php

`

**Comment by tarang777** [comment]
Hey jazgraphy .
Price inclusive of tax in woocommerce means at the time of creation of product u are specifying the actual price of product as well as the tax both together ...
You should try the option which is i will enter prices exclusive of taxes and then u create a tax class and assign to the product ok.

**Comment by jazgraphy** [comment]
Hi @tarang777,

Thanks for your reply.

> Price inclusive of tax in woocommerce means at the time of creation of product u are specifying the actual price of product as well as the tax both together ...

Yes, this is what I'm trying to achieve. 
I'm adding a product with RM100 (price inclusive of 6% tax).
Malaysia customer should pay 6% tax (RM94.34 + 6% tax = RM100), while other countries customer should see RM94.34. 

> You should try the option which is i will enter prices exclusive of taxes and then u create a tax class and assign to the product ok.

I'm importing my product list from somewhere else which is really huge (17k+ items) and the prices are inclusive of tax, it is very hard for me to re-enter the prices exclusive tax

Is there any other way I can solve this?
Thanks!

**Comment by mikejolley** [comment]
Base country should be set to Malaysia
Prices display in catalog/cart should be set to inclusive
Default customer location should be set to base country

Anyone outside of Malaysia will then see the tax-less price.

**Comment by jazgraphy** [comment]
Hi @mikejolley 
My base location is already set to Malaysia, default customer location also set to base country.
Prices are set to inclusive too.

Base location : Malaysia — Kuala Lumpur
Selling location(s) : Sell to all countries
Shipping location(s) : Ship to all countries you sell to
Default customer location : Shop base address
Enable taxes: Enable taxes Enable taxes and tax calculations

But the price is still calculate wrongly, any idea?

Thanks!

**Comment by mikejolley** [comment]
Screenshots?

**Comment by jazgraphy** [comment]
Here u go. 
[General Options](https://prnt.sc/fg4fdv)
[Tax](https://prnt.sc/fg4fg6)
[Standard rate](https://prnt.sc/fg4fok)

Thanks!

**Comment by mikejolley** [comment]
Try setting the state on the tax rate so it's for Kuala Lumpur?

**Comment by jazgraphy** [comment]
Hi @mikejolley ,

I realized the state code / post code / city is the one that causing the issue. 
It couldn't support the calculation if I added state code / postcode.

Now, I just set standard tax rate using country code MY
Using Country code only: 
![screenshot-barbersupply com my 2017-06-05 21-56-04](https://cloud.githubusercontent.com/assets/15964081/26786842/77366f48-4a3a-11e7-98df-34391f6a90a0.png)

Checkout page shows correct calculation for MY: 
![screenshot-barbersupply com my 2017-06-05 21-59-12](https://cloud.githubusercontent.com/assets/15964081/26786829/694766bc-4a3a-11e7-95ac-f72a123b983c.png)

Checkout page shows correct calculation for other countries:
![screenshot-barbersupply com my 2017-06-05 22-01-46](https://cloud.githubusercontent.com/assets/15964081/26786879/98ad8fa8-4a3a-11e7-918d-f6a3f0888cfe.png)

But, I need to set my standard tax rate according to different state & postcode, which caused calculation error. 
Postcode 87000...87100; 07000 should be tax free in MY, while other states should have 6% GST.
![screenshot-barbersupply com my 2017-06-05 20-51-21](https://cloud.githubusercontent.com/assets/15964081/26786975/ede7a0f8-4a3a-11e7-9777-2f2d321d17b5.png)
